sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java

```diff
@@ -77,13 +96,41 @@ private WriteResult(
       PCollection<TableRow> failedInserts,
       TupleTag<BigQueryInsertError> failedInsertsWithErrTag,
       PCollection<BigQueryInsertError> failedInsertsWithErr,
-      PCollection<TableRow> successfulInserts) {
+      PCollection<TableRow> successfulInserts,
+      TupleTag<TableDestination> successfulInsertsTag,
+      PCollection<TableDestination> successfulBatchInserts) {
     this.pipeline = pipeline;
     this.failedInsertsTag = failedInsertsTag;
     this.failedInserts = failedInserts;
     this.failedInsertsWithErrTag = failedInsertsWithErrTag;
     this.failedInsertsWithErr = failedInsertsWithErr;
     this.successfulInserts = successfulInserts;
+    this.successfulBatchInsertsTag = successfulInsertsTag;
+    this.successfulBatchInserts = successfulBatchInserts;
+  }
+
+  /**
+   * Returns a {@link PCollection} containing the {@link TableDestinations}s that were successfully
+   * inserted.
+   *
+   * <p>Successful Inserts are only produced when using batch inserts.
+   */
+  public PCollection<TableDestination> getSuccessfulBatchInserts() {
+    checkArgument(
+        successfulBatchInsertsTag != null,
+        "Cannot use getSuccessfulInserts because this WriteResult was not configured "
+            + "to produce them.  Note: streaming inserts do not produce successful "
+            + "insert results.");
+
+    return successfulBatchInserts;
```

Comment:
I just wonder about the name here. Maybe instead of  `getSuccessfulBatchInserts`  we can call it `getSuccessfulBatchWrites` ? IDK. WDYT?

Reply:
`getSuccessfulFileLoads` maybe to be consistent with the enum name?

Reply:
how about `getSuccessfulTableLoads`?

Reply:
Love it!